### ISSUE TITLE
Adding effects to played video

### ISSUE THREAD

**Issue by Rakatan** [brianwernick/ExoMedia#362]
Hello,

I have used your library so far with great success, so a heartfelt "Thank You!" would be a good way to start.

Onto business however: Is there any way to apply effects to the video that is being played? (Eg Sepia, Monochrome, etc...)
I have been looking through the source code and found no place where i could highjack the surface and apply a shader to it.

The only vaguely relevant search result i have come across would be [here](https://github.com/google/ExoPlayer/issues/1220), but no real solution was provided, or any sort of sample code for that matter.

Any insight would be greatly appreciated.

**Comment by brianwernick** [comment]
The `EMVideoView` uses a `TextureView` as the default backing type (can be changed to a `SurfaceView` or custom backing). You will need to extend `EMVideoView` to have access but you can use the [videoViewImpl](https://github.com/brianwernick/ExoMedia/blob/master/library/src/main/java/com/devbrackets/android/exomedia/ui/widget/EMVideoView.java#L82) and cast it to a [ResizingTextureView](https://github.com/brianwernick/ExoMedia/blob/master/library/src/main/java/com/devbrackets/android/exomedia/core/video/ResizingTextureView.java) or [ResizingSurfaceView](https://github.com/brianwernick/ExoMedia/blob/master/library/src/main/java/com/devbrackets/android/exomedia/core/video/ResizingSurfaceView.java) and access the surface that way.

**Comment by Rakatan** [comment]
Thank you for the quick response @brianwernick , i was unfortunately unable to respond that fast.

I have tried your suggestion and was indeed able to acquire the surface.
I then proceeded to go about my task in two ways:

1) Follow [this](https://www.virag.si/2014/03/rendering-video-with-opengl-on-android/) tutorial and replace the media player with my extended `EMVideoView`. I provided the renderer with the required surface, and while i can hear to sound playing, no image is displayed (even with a pass-through fragment shader).

2) I attempted to modify the `ResizingTextureView`, extending it from `GLSurfaceView` instead. Doing this allowed me to send said SurfaceView to a GPUImageInstance, much like in the [issue](https://github.com/google/ExoPlayer/issues/1220) i linked in my original post. Unfortunately i have encountered the exact same results, and was unable to proceed further.

So far your insight was very helpful, any other pointers would be great :).

>
